### PR TITLE
Removed Application from Ice Services

### PR DESCRIPTION
**Pull request by InsertCreativityHere**
This pr resolves issue #50.

### CODE REVIEW

File: cpp/src/IceBox/Admin.cpp [hunk 1]
```diff
@@ -7,44 +7,75 @@
...
+
+    try
+    {
+        //
+        // CtrlCHandler must be created before the communicator or any other threads are started
+        //
+        Ice::CtrlCHandler ctrlCHandler;
+
+        //
+        // CommunicatorHolder's ctor initializes an Ice communicator,
```
Comment (bernardnormier) [inline cpp/src/IceBox/Admin.cpp]: You should remove these comments, it's not a demo program.

File: cpp/src/IceBox/Admin.cpp [hunk 2]
```diff
@@ -67,18 +98,18 @@ Client::run(int argc, char* argv[])
...
     {
-        commands = opts.parse(argc, const_cast<const char**>(argv));
+        commands = opts.parse(args);
     }
     catch(const IceUtilInternal::BadOptException& e)
     {
         consoleErr << e.reason << endl;
-        usage();
+        usage(args[0]);
         return EXIT_FAILURE;
```
Comment (bernardnormier) [inline cpp/src/IceBox/Admin.cpp]: Let's use 0/1 or EXIT_SUCCESS/EXIT_FAILURE consistently - not a mix.

File: cpp/src/IceGrid/Client.cpp [hunk 3]
```diff
@@ -113,206 +109,189 @@ class SessionKeepAliveThread : public IceUtil::Thread, public IceUtil::Monitor<I
...
-    IceUtil::CtrlCHandler _ctrlCHandler;
-    CommunicatorPtr _communicator;
-    string _appName;
-    ParserPtr _parser;
-};
+//
+// Global variables for interruptCallback
+//
+Ice::CommunicatorPtr communicator;
+IceGrid::ParserPtr parserPtr;
```
Comment (bernardnormier) [inline cpp/src/IceGrid/Client.cpp]: The parser's name should be simply parser

File: cpp/src/IceGrid/Client.cpp [hunk 4]
```diff
@@ -113,206 +109,189 @@ class SessionKeepAliveThread : public IceUtil::Thread, public IceUtil::Monitor<I
...
-
-    IceUtil::CtrlCHandler _ctrlCHandler;
-    CommunicatorPtr _communicator;
-    string _appName;
-    ParserPtr _parser;
-};
+//
+// Global variables for interruptCallback
+//
+Ice::CommunicatorPtr communicator;
```
Comment (bernardnormier) [inline cpp/src/IceGrid/Client.cpp]: These variables should be next to the static mutex.

File: cpp/src/IceBox/Admin.cpp [hunk 5]
```diff
@@ -7,44 +7,75 @@
...
+        // Destroy communicator on Ctrl-C
+        //
+        ctrlCHandler.setCallback(&destroyCommunicator);
+
+        Ice::StringSeq args = Ice::argsToStringSeq(argc, argv);
+        status = run(args);
+    }
+    catch(const std::exception& ex)
+    {
+        cerr << ex.what() << endl;
```
Comment (pepone) [inline cpp/src/IceBox/Admin.cpp]: You must use consoleErr, this ensure message gets converted to the console encoding

File: cpp/src/IceBox/Admin.cpp [hunk 6]
```diff
@@ -67,18 +98,18 @@ Client::run(int argc, char* argv[])
     vector<string> commands;
     try
     {
-        commands = opts.parse(argc, const_cast<const char**>(argv));
+        commands = opts.parse(args);
```
Comment (bernardnormier) [inline cpp/src/IceBox/Admin.cpp]: I think it would be more logical to put the result in args. Otherwise, pass args as a const&.

**Pull request merged**